lib/semigrp.gi

```diff
@@ -107,15 +107,9 @@ function(S)
     suffix := Concatenation("\>of\< ", suffix);
   fi;
   Append(str, suffix);
-
-  Append(str, "\>with\< \>");
-  Append(str, ViewString(nrgens));
-  Append(str, "\< \>generator");
-  if nrgens > 1 or nrgens = 0 then
-    Append(str, "s");
-  fi;
-  Append(str, "\<>\<");
-
+  Append(str, "\>with\< ");
+  Append(str, Pluralize(nrgens, "generator"));
+  Append(str, ">\<");
```

Comment:
The number of opening `\>` and closing `\<` doesn't match.

Reply:
The number does match, by looking at the function `_ViewStringForSemigroups` as a whole.

* Initially, on line 40, `str` is defined to be a string with one `\>` control character.
* In the lines immediately before `str` is returned (i.e. on lines 79 and 112), the corresponding additional `\<` is added.
* In all other places, each 'block' of code adds an equal number of `\>` and `\<` characters.
    * In particular, on each line where control characters are added, just a single `\>` and a single `\<` are added, with the exception of the lines immediately before `return str;` (as already mentioned) and with the exception of lines 98–100, where this is done slightly differently.

I hope so, anyway (I find it really hard to distinguish between the two characters...)